File: TravelAgencyRestApi/src/main/java/com/travelagency/travelagencyapplication/dto/response/ClientResponse.java

package com.travelagency.travelagencyapplication.dto.response;

import com.travelagency.travelagencyapplication.collection.Passport;
import com.travelagency.travelagencyapplication.enums.Role;
import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ClientResponse extends UserResponse{
    private Passport passport;
    private List<TravelResponse> clientTravels;
}

File: TravelAgencyFrontend/src/app/types/Client/ClientResponse.ts
import {User} from "../User/User";
import {Passport} from "./Passport";
import {ReservationResponse} from "../Reservation/ReservationResponse";
import {UserResponse} from "../User/UserResponse";

export interface ClientResponse extends UserResponse{
  passport:Passport;
  reservations?:ReservationResponse[];
}
